akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -410,8 +410,16 @@ private[http] object HttpServerBluePrint {
               openRequests = openRequests.enqueue(r)
               messageEndPending = r.createEntity.isInstanceOf[StreamedEntityCreator[_, _]]
               val rs = if (r.expect100Continue) {
-                oneHundredContinueResponsePending = true
-                r.copy(createEntity = with100ContinueTrigger(r.createEntity))
+                r.createEntity match {
+                  case StrictEntityCreator(HttpEntity.Strict(_, _)) =>
```

Comment:
Before the git force push,  this line was:
```scala
case StrictEntityCreator(HttpEntity.Strict(_, ByteString.empty)) =>
```
As you can see I checked only for an empty `ByteString`. However as soon there is a strict entity it means we are done already and no data are expected anymore.

Reply:
:+1: thanks for the source comments.